apps/system/lockscreen/js/lockscreen_state_manager.js

```diff
@@ -415,10 +416,52 @@
         this.onUnlockingApp();
         break;
       case 'lockscreenslide-activate-right':
+        isFromNotification = false;
+        appID = '';
+        this.onActivateUnlock();
+        break;
       case 'lockscreen-notification-request-activate-unlock':
         this.onActivateUnlock();
+        isFromNotification = true;
+        appID = detail.id;
         break;
       case 'lockscreen-request-unlock':
+      // if app is launched from the lockscreen notification screen then it will
+      // find the that app in system and launch it.
+        if (isFromNotification) {
+          isFromNotification = false;
+          var request = navigator.mozApps.mgmt.getAll();
```

Comment:
Even if this was the right way to launch apps, isn't there a huge security hole here? What if the user has specified a PIN. Don't you need to check that the pin has been entered before launching anything?